index.d.ts

```diff
@@ -23,16 +23,29 @@ export interface PoolConfig extends mysql.PoolConfig {
     reconnect?: boolean;
 }
 
-export interface QueryFunction<T> {
-    (query: mysql.Query | string | mysql.QueryOptions): T;
+export interface QueryFunction {
+    <T extends unknown>(query: mysql.Query | string | mysql.QueryOptions): T;
```

Comment:
I'm not sure I understand what extending unknown does here, could you explain it please?

Reply:
I was dealing with a case just like this earlier in the week. I'm pretty sure you want `T = unknown` here, rather than `T extends unknown`.

According to [@typescript-eslint/no-unnecessary-type-constraint](https://typescript-eslint.io/rules/no-unnecessary-type-constraint):

> the Constraining the generic type `T` to `unknown` does nothing and is unnecessary

Another question we have to consider is what developers are going to be expecting when they don't provide a type. For my own projects I would agree with defaulting to `unknown`, but I think many developers would be expecting `any`. If we default to `unknown`, it will break our consumers' existing code, while defaulting to `any` shouldn't cause any issues. (For reference, RxJS defaults to `unknown`, but axios defaults to `any`).

Reply:
_"Even without T = unknown, the return type will default to unknown anyway"_ that is true but only if you are using newer versions of typescript.

As for  ```unknown``` vs ```any``` I would prefer ```unknown``` but I agree that for compatibility reasons ```any``` is more suitable (except if the major version is bumped 😃 )

I could just refactor this to ```any``` and  problem is solved. What do you think?

Reply:
I think we should err on the side of compatibility rather than strictness here. I'm thinking we should default to `any` with `T = any`. That would allow developers to keep using their existing code without changes, but also give them the flexibility to use this new functionality if they want to.

Reply:
I agree too. I pushed the changes